packages/uxpin-merge-cli/src/program/command/push/steps/uploadLibrary.ts

```diff
@@ -102,12 +133,34 @@ export function uploadLibrary(buildOptions:BuildOptions):StepExecutor {
       printLine(`✅ Projects set to use DS branch [${vcsDetails.branchName}]!`, { color: PrintColor.GREEN });
     } catch (error) {
       printLine(
-        `🛑 There was an error while updating design system pointers [${vcsDetails.branchName}] .`,
+        `🛑 There was an error while updating design system pointers [${vcsDetails.branchName}]`,
         { color: PrintColor.RED },
       );
       throw new Error(error.message);
     }
 
+    if (tag) {
+      try {
+        await createTag({
+          apiDomain,
+          authToken,
+          commitHash,
+          tag,
+        });
+
+        printLine(
+          `🏷️  Library tagged at this point in time with tag [${tag}] at commit hash [${commitHash}]`,
+          { color: PrintColor.YELLOW },
+        );
+      } catch (error) {
+        printLine(
+          `🛑 There was an error while creating a tag [${tag}] at commit hash [${commitHash}]`,
+          { color: PrintColor.RED },
+        );
+        throw new Error(error.message);
+      }
+    }
```

Comment:
I see the same code above. Maybe better to create a helper function instead of repeating code?